File: Frontend/src/models/module.ts
import { Menu } from './menu';
import { Functionality } from 'models/functionality';
export class Module{
    constructor(
        public id: number,
        public code: string,
        public description: string,
        public active: boolean,
        public menuId: number,
        public menu: Menu,
        public functionalities: Functionality[] = null,
        public available: boolean = false
    ){}
}

File: src/Sofco.Model/Models/Admin/Module.cs
using System.Collections.Generic;
using Sofco.Common.Domains;

namespace Sofco.Domain.Models.Admin
{
    public class Module : BaseEntity, ILogicalDelete
    {
        public string Description { get; set; }

        public bool Active { get; set; }

        public string Code { get; set; }

        public IList<Functionality> Functionalities { get; set; }
    }
}
